### PR TITLE
Move TALK_COMMON_ALLY options into submenus

### PR DESCRIPTION
**Pull request by I-am-Erk**
#### Summary
SUMMARY: Content "Rearrange Ally conversation menu to use submenus"

#### Purpose of change
The NPC follower conversation window is getting crowded:
image

The list is getting to the point that specific options are hard to catch on the first, or even third read-through. Plus, we are running out of letters.

#### Describe the solution
Creates several submenus to hold options.
- the first two options remain in the root menu because they're likely the most common things people talk to npcs about. I would suggest that these should be taken out of dialogue entirely, and become separate UIs that you can access with `C` or by bumping into the NPC. But that's a PR for another day.
- "chat" contains backstories, small talk, and NPC missions.
- "orders" contains nearly everything else you can get from an NPC, from inventory management to teaching you things. It also includes the camp orders stuff, although that also still shows up in the "tell me about faction camps" area.
- "help" contains NPC help files eg. setting up faction camps.

I may add a "join this faction camp" order to Orders in addition to the implicit one within 'tell me about faction camps'.

#### Describe alternatives you've considered
It could probably stay as it is for a while yet, but it's better to get something like this done sooner and just pull off the bandaid before our motor memories get even more set in their ways.

I'm open to alternative organization suggestiosn.

#### Testing
image
image

#### Additional context
Once this is done I plan to expand some of the 'chat' options.

### CODE REVIEW

File: data/json/npcs/TALK_COMMON_ALLY.json [hunk 1]
```diff
@@ -46,7 +46,71 @@
...
       { "text": "I want to give you some commands for combat.", "topic": "TALK_COMBAT_COMMANDS" },
       { "text": "I want to set some miscellaneous rules.", "topic": "TALK_MISC_RULES" },
+      {
+        "text": "I'd like to know a bit more about your abilities.",
+        "topic": "TALK_FRIEND",
+        "effect": "reveal_stats"
+      },
+      { "text": "There's something I want you to do.", "topic": "TALK_ALLY_ORDERS" },
+      { "text": "I just wanted to talk for a bit.", "topic": "TALK_ALLY_SOCIAL" },
+      { "text": "Can you help me understand something?", "topic": "TALK_ALLY_HELP" },
```
Comment (mlangsdorf) [inline data/json/npcs/TALK_COMMON_ALLY.json:56]:
```suggestion
      { "text": "Can you help me understand something? (NPC TUTORIAL)", "topic": "TALK_ALLY_HELP" },
```

Please make it obvious that there is an NPC tutorial in the top level menu, for what little good it does.

Comment (I-am-Erk) [reply data/json/npcs/TALK_COMMON_ALLY.json:56]: It might do more good if there aren't 20 options to read through

Comment (mlangsdorf) [reply data/json/npcs/TALK_COMMON_ALLY.json:56]: It still needs to be as obvious as possible.

Comment (I-am-Erk) [reply data/json/npcs/TALK_COMMON_ALLY.json:56]: I agree just haven't been near a computer to fix it for a bit yet

**Pull request merged**